tools/core-build-tasks/src/tasks/bundle.ts

```diff
@@ -17,20 +21,105 @@ export type BundleTaskParameters = {
     /** The output file for the bundle. Documentation: https://esbuild.github.io/api/#outfile */
     outfile: string;
 
-    /** Flag to specify how to generate source map. Documentation: https://esbuild.github.io/api/#sourcemap */
-    sourcemap?: boolean | 'linked' | 'inline' | 'external' | 'both';
+    /** Flag to specify to generate a source map file.*/
+    sourcemap?: boolean;
+
+    /** The output path for the source map file. Only when sourcemap option is set to true. */
+    outputSourcemapPath?: string;
 };
 
+export type PostProcessOutputFilesResult = {
+    sourceMapDirectory: string;
+    outputDirectory: string;
+    generatedFiles: { [key: string]: string };
+};
+
+function linkSourceMaps(
+    sourceMapDirectory: string,
+    outputDirectory: string,
+    outputFiles: OutputFile[]
```

Comment:
Where are these outputFiles written to? Are they a temp directory? Should we clean them up after the copy?

Reply:
These are generated in memory, sending `write` flag to false in case of generating source maps.